compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1268,31 +1268,460 @@ OMR::ARM64::TreeEvaluator::mstoreiEvaluator(TR::Node *node, TR::CodeGenerator *c
 TR::Register*
 OMR::ARM64::TreeEvaluator::mTrueCountEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   TR::Node *firstChild = node->getFirstChild();
+   TR_ASSERT_FATAL_WITH_NODE(node, firstChild->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", firstChild->getDataType().toString());
+
+   TR::DataType et = firstChild->getDataType().getVectorElementType();
+   TR::Register *maskReg = cg->evaluate(firstChild);
+   TR::Register *tempReg = cg->allocateRegister(TR_VRF);
+   TR::Register *resReg = cg->allocateRegister();
+   TR::InstOpCode::Mnemonic negOp;
+   TR::InstOpCode::Mnemonic addvOp;
+
+   switch (et)
+      {
+      case TR::Int8:
+         negOp = TR::InstOpCode::vneg16b;
+         addvOp = TR::InstOpCode::vaddv16b;
+         break;
+      case TR::Int16:
+         negOp = TR::InstOpCode::vneg8h;
+         addvOp = TR::InstOpCode::vaddv8h;
+         break;
+      case TR::Int32:
+      case TR::Float:
+         negOp = TR::InstOpCode::vneg4s;
+         addvOp = TR::InstOpCode::vaddv4s;
+         break;
+      case TR::Int64:
+      case TR::Double:
+         negOp = TR::InstOpCode::vneg2d;
+         addvOp = TR::InstOpCode::vaddv4s; /* we do not have addv for long, but vaddv4s works for this case */
+         break;
+      default:
+         TR_ASSERT_FATAL_WITH_NODE(node, false, "Unexpected element type");
+         break;
+      }
+
+   generateTrg1Src1Instruction(cg, negOp, node, tempReg, maskReg);
+   generateTrg1Src1Instruction(cg, addvOp, node, tempReg, tempReg);
+   generateMovVectorElementToGPRInstruction(cg, TR::InstOpCode::umovwb, node, resReg, tempReg, 0);
+
+   node->setRegister(resReg);
+   cg->stopUsingRegister(tempReg);
+   cg->decReferenceCount(firstChild);
+
+   return resReg;
    }
 
 TR::Register*
 OMR::ARM64::TreeEvaluator::mFirstTrueEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   TR::Node *firstChild = node->getFirstChild();
+   TR_ASSERT_FATAL_WITH_NODE(node, firstChild->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", firstChild->getDataType().toString());
+
+   TR::DataType et = firstChild->getDataType().getVectorElementType();
+   TR::Register *maskReg = cg->evaluate(firstChild);
+   TR::Register *temp1Reg = cg->allocateRegister(TR_VRF);
```

Comment:
The name of the temporary register is `temp1Reg` in this function, `mLastTrueEvaluator()`, and `mToLongBitsEvaluator()`.  `mTrueCountEvaluator()` and `mLongBitsToMaskEvaluator()` uses `tempReg`.

Reply:
Changed the name of the variable to `tempReg`.